From: Johannes Thumshirn <morbidrsa@googlemail.com>
Date: Mon, 6 Aug 2012 14:08:50 +0200
Subject: [040/106] staging: line6: pcm.c: Changed simple_strtoul to kstrtoint

commit cdf5e55124e4804fc0027df7e89567a3f5eed8bf upstream.

Changed call to simple_strtoul to kstrtoint in pcm_set_impulse_volume(...)

Signed-off-by: Johannes Thumshirn <morbidrsa@googlemail.com>
Reviewed-by: Stefan Hajnoczi <stefanha@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/line6/pcm.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

Index: linux-3.2.46/drivers/staging/line6/pcm.c
===================================================================
--- linux-3.2.46.orig/drivers/staging/line6/pcm.c	2013-07-26 19:28:35.000000000 +0000
+++ linux-3.2.46/drivers/staging/line6/pcm.c	2013-07-26 19:28:50.000000000 +0000
@@ -48,7 +48,13 @@
 				      const char *buf, size_t count)
 {
 	struct snd_line6_pcm *line6pcm = dev2pcm(dev);
-	int value = simple_strtoul(buf, NULL, 10);
+	int value;
+	int rv;
+
+	rv = kstrtoint(buf, 10, &value);
+	if (rv < 0)
+		return rv;
+
 	line6pcm->impulse_volume = value;
 
 	if (value > 0)
